src/sys/wait.rs

```diff
@@ -215,24 +232,53 @@ fn decode(pid : Pid, status: i32) -> WaitStatus {
     }
 }
 
-pub fn waitpid<P: Into<Option<Pid>>>(pid: P, options: Option<WaitPidFlag>) -> Result<WaitStatus> {
+/// Waits for and returns events that are received from the given supplied process or process group
+/// ID, and associated options.
+///
+/// # Usage Notes
+///
+/// - If the value of PID is greater than `0`, it will wait on the child with that PID.
+/// - If the value of the PID is less than `-1`, it will wait on any child that
+///   belongs to the process group with the absolute value of the supplied PID.
+/// - If the value of the PID is `0`, it will wait on any child process that has the same
+/// group ID as the current process.
+/// - If the value of the PID is `-1`, it will wait on any child process of the current process.
+/// - If the value of the PID is `None`, the value of PID is set to `-1`.
```

Comment:
I think that'd be a much better option, although it would require a version bump due to breaking the existing API. I think the best would be to have the parameter value as `Into<PidGroup>`, where `PidGroup` could also implement `From<i32>`, so that we could continue using `i32` values as in C in addition.